baybe/recommenders/pure/bayesian/botorch.py

```diff
@@ -152,25 +155,132 @@ def _recommend_continuous(
 
         import torch
         from botorch.optim import optimize_acqf
-
-        points, _ = optimize_acqf(
-            acq_function=self._botorch_acqf,
-            bounds=torch.from_numpy(subspace_continuous.param_bounds_comp),
-            q=batch_size,
-            num_restarts=5,  # TODO make choice for num_restarts
-            raw_samples=10,  # TODO make choice for raw_samples
-            equality_constraints=[
-                c.to_botorch(subspace_continuous.parameters)
-                for c in subspace_continuous.constraints_lin_eq
-            ]
-            or None,  # TODO: https://github.com/pytorch/botorch/issues/2042
-            inequality_constraints=[
-                c.to_botorch(subspace_continuous.parameters)
-                for c in subspace_continuous.constraints_lin_ineq
-            ]
-            or None,  # TODO: https://github.com/pytorch/botorch/issues/2042
-            sequential=self.sequential_continuous,
-        )
+        from torch import Tensor
```

Comment:
Agree and thanks for the suggestion. I change it now to a factory method. In `_recommend_continuous_with_cardinality_constraints`, I'm still using a closure instead of a helper function in to avoid having too many arguments in its signature.

Reply:
Thx for the changes. I've finally had some to start the actual review, so I'll close here and give you some more comments in separate threads